cmd/livepeer/livepeer.go

```diff
@@ -639,6 +639,20 @@ func main() {
 
 		bcast := core.NewBroadcaster(n)
 
+		var dbOrchPoolCache *discovery.DBOrchestratorPoolCache
+		// When the node is on-chain mode always cache the on-chain orchestrators and poll for updates
+		// Right now we rely on the DBOrchestratorPoolCache constructor to do this. Consider separating the logic
```

Comment:
Keep in mind that even if we seperate this logic (which I'm in favor of) we would still be processing events for all O's even those we don't care about at startup, this is also true when starting the node as an orchestrator currently. Perhaps seperating these concerns depending on the discovery method chosen can be extracted into a seperate issue though.

Reply:
> Perhaps seperating these concerns depending on the discovery method chosen can be extracted into a seperate issue though.

Something else to consider is validation of the orchs passed in via `-orchAddr` or `-orchWebhookURL` when the node is on-chain mode to ensure that they are active on-chain.

Let's discuss in a separate issue.